core/include/detray/utils/algebra_helpers.hpp

```diff
@@ -0,0 +1,60 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+#include "detray/definitions/qualifiers.hpp"
+
+namespace detray {
+
+template <typename scalar_t>
+struct vector_helpers {
+
+    using vector3 = __plugin::vector3<scalar_t>;
+
+    using size_type = __plugin::size_type;
+
+    template <size_type ROWS, size_type COLS>
+    using matrix_type = __plugin::matrix_type<scalar_t, ROWS, COLS>;
+
+    using matrix_operator = standard_matrix_operator<scalar_t>;
+
+    /// Column-wise cross product between matrix (m) and vector (v)
+    DETRAY_HOST_DEVICE
+    inline matrix_type<3, 3> cross(const matrix_type<3, 3>& m,
```

Comment:
Just a dumb question, but should these not go into algebra-plugins?

Reply:
Maybe not because these column-wise cross and dot products are not _official_ matrix operations.

Reply:
Hm, it seems Eigen at least has more native implementations to do this. Like ```Eigen::VectorwiseOp``` Have you considered those?